### PR TITLE
cleanup old not functioning epel and yumrepo variables from the redhat o...

### PR DESCRIPTION
**Pull request by abenari**
...s family code

replacing it with a place holder for repos, that can be override by a plugin.

### CODE REVIEW

File: app/models/operatingsystems/redhat.rb [hunk 1]
```diff
@@ -17,29 +17,9 @@ def mediumpath host
...
-      epel_url.gsub!("$os","4-10")
-    when "5"
-      epel_url.gsub!("$os","5-4")
-    when "6"
-       epel_url.gsub!("$os","6-8")
-    else
-      return ""
-    end
-    "su -c 'rpm -Uvh #{medium_uri(host, epel_url)}'"
+  def repos host
```
Comment (domcleal) [inline app/models/operatingsystems/redhat.rb]: The RedHat family seems the wrong place for it, is it not more universal than that?

**Comment by domcleal** [comment]
The interface for 'repos' seems ill-defined to me, can we state what sort of data is expected or provide more structure around it than just expecting a plugin to override it?

A redmine ticket is also required explaining what's being changed + why, thanks.

**Comment by abenari** [comment]
@domcleal I see your point about the interface definition, how do you suggest defining the interface?

**Comment by domcleal** [comment]
@abenari at the very simplest a comment so we know whether it's passing back an array of hashes or whatever, but I think we should probably look to something like YARD to begin documenting extension points that we expect people to override or not (e.g. `@api public`).

How are you overriding it at the moment in the plugin?  By changing the repos method in the OS model when your plugin's loaded?

**Comment by abenari** [comment]
@domcleal added documentation in the method and renamed the commit to point to the the redmine issue.

**Comment by domcleal** [comment]
@abenari that looks good, thanks.

I see `epel` in Operatingsystem::Jail (app/models/operatingsystem.rb:54), I guess this needs changing to `repos` so you're able to access this helper in safemode?

**Comment by domcleal** [comment]
Merged as 6b27556 with a typo fix, thanks @abenari.

**Pull request closed without merging**